build-system/tasks/extension-helpers.js

```diff
@@ -561,18 +563,54 @@ async function buildExtensionJs(path, name, version, latestVersion, options) {
   }
 
   if (name === 'amp-script') {
-    // Copy @ampproject/worker-dom/dist/amp/worker/worker.js to dist/ folder.
-    const dir = 'node_modules/@ampproject/worker-dom/dist/amp/worker/';
-    const file = `dist/v0/amp-script-worker-${version}`;
-    // The "js" output is minified and transpiled to ES5.
-    fs.copyFileSync(dir + 'worker.js', `${file}.js`);
-    // The "mjs" output is unminified ES6 and has debugging flags enabled.
-    fs.copyFileSync(dir + 'worker.mjs', `${file}.max.js`);
-
-    // Same as above but for the nodom worker variant.
-    const noDomFile = `dist/v0/amp-script-worker-nodom-${version}`;
-    fs.copyFileSync(dir + 'worker.nodom.js', `${noDomFile}.js`);
-    fs.copyFileSync(dir + 'worker.nodom.mjs', `${noDomFile}.max.js`);
+    const workerDomDir = 'node_modules/@ampproject/worker-dom';
+    const newDirExists = fs.existsSync(`${workerDomDir}/dist/amp-production`);
+    const targetDir = 'dist/v0';
+
+    log('Copying required @ampproject/worker-dom files.');
+
+    if (newDirExists) {
+      const dir = `${workerDomDir}/dist`;
+      const workerFilesToDeploy = await globby(
+        `${dir}/amp-production/worker/*`,
+        `${dir}/amp-debug/worker/*`
+      );
+      for (const filepath of workerFilesToDeploy) {
```

Comment:
Optional: Do you think it could be simpler with a map instead? e.g.
```js
const filenameMap = {
  'amp-production/worker.js': `amp-script-worker-${version}.js`,
  'amp-production/worker.mjs': `amp-script-worker-${version}.mjs`,
  'amp-production/worker.nodom.js': `amp-script-worker-nodom-${version}.js`,
  'amp-production/worker.nodom.mjs': `amp-script-worker-nodom-${version}.mjs`,

  'amp-debug/worker.js': `amp-script-worker-${version}.max.js`,
  'amp-debug/worker.mjs': `amp-script-worker-${version}.max.mjs`,
  'amp-debug/worker.nodom.js': `amp-script-worker-nodom-${version}.max.js`,
  'amp-debug/worker.nodom.mjs': `amp-script-worker-nodom-${version}.max.mjs`,
}
```

Reply:
Agree, this would make the code more maintainable.

Reply:
i actually made this a map originally, if that is preferred i will do that. i simply switched to this since everything was copyable

Reply:
done.